src/v/cloud_storage/cache_service.cc

```diff
@@ -310,8 +314,8 @@ ss::future<> cache::trim(
     // from cache directory by the user manually.
     co_await _access_time_tracker.trim(candidates_for_deletion);
 
-    auto size_limit = size_limit_override.value_or(_max_bytes);
-    auto object_limit = object_limit_override.value_or(_max_objects());
+    auto size_limit = trim_target_bytes(size_limit_override);
+    auto object_limit = trim_target_objects(object_limit_override);
```

Comment:
I'm kind of surprised that this is being baked into `trim()`, instead of e.g. adding a background run of `trim()` where these configs are passed in as the override arguments when we call `do_reserve_space()`.

Even if `trim()` honors these lower values, don't we still need something to kick off the trim earlier?

Reply:
The trim is kicked of by space reservations. Once space reservation will hit the lower limits it will start trim in the background and since we're guaranteed to have space it will not block the reader that needed space.

Our problem is that in some cases the trim is slower than the cache ingestion rate. We start trim without blocking the readers but eventually, the readers fill up cache and the trim becomes blocking. This is why moving the threshold works in this case. This could also be implemented as an adaptive strategy. We can project the time when the cache will get full and how long we will need to trim and then start the trim early based on that information. But the simplest solution is to just add a config parameter.

Reply:
>The trim is kicked of by space reservations. Once space reservation will hit the lower limits it will start trim in the background and since we're guaranteed to have space it will not block the reader that needed space.

Right, but my point is that the condition before triggering a trim hasn't changed, if I'm reading correctly. `do_reserve_space()` returns early without trimming if `may_reserve_space()` is true, which only examines honors hard limits. This PR doesn't change that condition, it only reduces the count/size that we trim down to.

Reply:
maybe the code was changed since my previous comment but it looks like this new config triggers the background trim